API/src/main/java/de/brandwatch/minianalytics/api/controller/MentionController.java

```diff
@@ -2,64 +2,63 @@
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
 
-import java.time.Instant;
+import java.time.DateTimeException;
+import java.time.LocalDate;
 import java.time.LocalDateTime;
-import java.time.ZoneId;
-import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
-import java.time.temporal.TemporalAccessor;
-import java.util.List;
+import java.time.ZoneOffset;
 
 @RestController
 public class MentionController {
 
-    private static final Logger logger = LoggerFactory.getLogger(RestController.class);
+    private static final Logger logger = LoggerFactory.getLogger(MentionController.class);
 
     private static final Gson gson = new Gson();
 
-    private final MentionRepository mentionRepository;
+    private final MentionService mentionService;
 
     @Autowired
-    public MentionController(MentionRepository mentionRepository) {
-        this.mentionRepository = mentionRepository;
+    public MentionController(MentionService mentionService) {
+        this.mentionService = mentionService;
+        LocalDate.now().minusDays(1).toString();
     }
 
     @GetMapping(value = "/mentions/{queryID}")
     public ResponseEntity findMentionsFromQuery(
             @PathVariable String queryID,
-            @RequestParam(value = "date", defaultValue = "") String date){
-
-        logger.info("GET /mentions/" + queryID);
-
-        List<Mention> mentions = mentionRepository.findByQueryID(Long.parseLong(queryID));
-
-        if(!date.equals("")){
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-
-            String convertedDate = date + " 00:00:00";
-            TemporalAccessor temporalAccessor = formatter.parse(convertedDate);
-            LocalDateTime localDateTime = LocalDateTime.from(temporalAccessor);
-            ZonedDateTime zonedDateTime = ZonedDateTime.of(localDateTime, ZoneId.systemDefault());
-            Instant instant = Instant.from(zonedDateTime);
-
-            String dateBounds = "[" + instant + " TO *]";
-
-            logger.info("queryID: *?0* AND date: *?1*");
-
-            return ResponseEntity.status(200).body(gson.toJson(mentionRepository.findMentionsAfterDate(
-                    Long.parseLong(queryID), dateBounds)));
+            @RequestParam(value = "startDate", defaultValue = "")
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime startDate,
+            @RequestParam(value = "endDate", defaultValue = "")
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime endDate) {
+        try{
```

Comment:
There's a more generic way to handle exceptions in spring framework with ControllerAdvice or exception handlers, e.g. like here https://medium.com/@jovannypcg/understanding-springs-controlleradvice-cd96a364033f

Do you want to try it or would you like to leave it like it is for now?

Reply:
Actually that (https://medium.com/@jovannypcg/understanding-springs-controlleradvice-cd96a364033f) sounds pretty logical, but I don't know all the exact Exceptions that can be thrown in these methods. I could ofc. write a ControllerAdvice catching the general `Exception.class`, but I don't think that would be very helpful.